script.js

```diff
@@ -0,0 +1,368 @@
+  KeyH: ['h', 'h', 'H', 'р', 'Р'],
+  KeyJ: ['j', 'j', 'J', 'о', 'О'],
+  KeyK: ['k', 'k', 'K', 'л', 'Л'],
+  KeyL: ['l', 'l', 'L', 'д', 'Д'],
+  Semicolon: ['colon', ';', ':', 'ж', 'Ж'],
+  Quote: ['quotationMarks', "'", '"', 'э', 'Э'],
+  Enter: ['enter', '\n', '\n', '\n', '\n'],
+  ShiftLeft: ['shift'],
+  KeyZ: ['z', 'z', 'Z', 'я', 'Я'],
+  KeyX: ['x', 'x', 'X', 'ч', 'Ч'],
+  KeyC: ['c', 'c', 'C', 'с', 'С'],
+  KeyV: ['v', 'v', 'V', 'м', 'М'],
+  KeyB: ['b', 'b', 'B', 'и', 'И'],
+  KeyN: ['n', 'n', 'N', 'т', 'Т'],
+  KeyM: ['m', 'm', 'M', 'ь', 'Ь'],
+  Comma: ['comma', ',', '<', 'б', 'Б'],
+  Period: ['point', '.', '>', 'ю', 'Ю'],
+  Slash: ['questionMark', '/', '?', '.', ','],
+  ArrowUp: ['arrowUp', '▲', '▲', '▲', '▲'],
+  ShiftRight: ['rShift'],
+  ControlLeft: ['ctrl'],
+  MetaLeft: ['win'],
+  AltLeft: ['alt'],
+  Space: ['space', ' ', ' ', ' ', ' '],
+  AltRight: ['rAlt'],
+  ControlRight: ['rCtrl'],
+  ArrowLeft: ['arrowLeft', '◄', '◄', '◄', '◄'],
+  ArrowDown: ['arrowDown', '▼', '▼', '▼', '▼'],
+  ArrowRight: ['arrowRight', '►', '►', '►', '►'],
+};
+
+if (sessionStorage.lang === undefined) sessionStorage.lang = 'ENG';
+let lastPressButton = '';
+let shiftCase = false;
+let shiftCaseLeft = false;
+let altCaseLeft = false;
+let changeLangUpKeyFlag = false;
+let capsLock = false;
+
+function createVirtualKeyboard() {
+  const wrapper = document.createElement('div');
+  wrapper.className = 'wrapper';
+  document.body.append(wrapper);
+
+  const textarea = document.createElement('textarea');
+  textarea.className = 'textarea';
+  wrapper.append(textarea);
+
+  const keyboard = document.createElement('div');
+  keyboard.className = 'keyboard';
+  wrapper.append(keyboard);
+
+  const clearfix = document.createElement('div');
+  clearfix.className = 'clearfix';
+  keyboard.append(clearfix);
+
+  keyboard.after("Shift + alt - change language");
+}
+
+createVirtualKeyboard();
+
+const keyboard = document.querySelector('.keyboard');
+const textarea = document.querySelector('.textarea');
+const clearfix = document.querySelector('.clearfix');
+
+class Button {
+  constructor({ code, key, style }) {
+    this.className = `button ${style}`;
+    this.code = code;
+    this.key = key;
+  }
+
+  buildButton() {
+    const div = document.createElement('div');
+    div.className = this.className;
+    div.dataset.key = this.key;
+    div.dataset.code = this.code;
+    div.innerHTML = this.code;
+    return div;
+  }
+}
+
+function creatButton() {
+  const ar = [];
+  const arr = Object.keys(keyboardElements);
+  arr.forEach((el) => {
+    ar.push(
+      new Button({
+        code: el,
+        key: keyboardElements[el][0],
+        style: keyboardElements[el][0],
+      })
+    );
+  });
+  ar.forEach((el) => {
+    keyboard.append(el.buildButton());
+  });
+}
+
+creatButton();
+showKey();
+textarea.focus();
+
+function showKey() {
+  let arrValues = Object.values(keyboardElements);
+
+  arrValues.forEach((element) => {
```

Comment:
Also don't see the reason for this code. Can you explain?

Reply:
Fix